File: 739-daily-temperatures/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {

        if (temperatures.empty())
          return {};

        int n = temperatures.size();
        vector<int> res(n, 0);
 
        stack<int> st;
        for (int i = 0 ; i < n; i++)
        {
           while(!st.empty() && temperatures[i] > temperatures[st.top()] )
           {
              int preIdx = st.top();
              st.pop(); 

              res[preIdx] = i - preIdx;
           }

           st.push(i);
        }
        
        return res;
    }
};

File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        
        int n = temperatures.length;

        Stack<Integer> st = new Stack<Integer>();
        int[] res =  new int[n];

        for (int i = 0; i < n; i++)
        {
          while (!st.isEmpty() && temperatures[i] > temperatures[st.peek()])
          {
            int pre_idx = st.peek();
            st.pop();
            
            res[pre_idx] = i -  pre_idx;
          }

          st.push(i);
        }

        return res;
    }
}